test/ech_test.c

```diff
@@ -34,6 +558,312 @@ const OPTIONS *test_get_options(void)
+        TEST_info("OSSL_ECHSTORE_get1_info unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_downselect(NULL, 0), 0)) {
+        TEST_info("OSSL_ECHSTORE_downselect unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_downselect(es, 100), 0)) {
+        TEST_info("OSSL_ECHSTORE_downselect unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_set1_key_and_read_pem(NULL, priv,
+                                                         inout, 0), 0)) {
+        TEST_info("OSSL_ECHSTORE_set1_key_and_readp_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_set1_key_and_read_pem(es, NULL,
+                                                         inout, 0), 0)) {
+        TEST_info("OSSL_ECHSTORE_set1_key_and_readp_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_set1_key_and_read_pem(es, priv,
+                                                         NULL, 0), 0)) {
+        TEST_info("OSSL_ECHSTORE_set1_key_and_readp_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_set1_key_and_read_pem(es, priv,
+                                                         inout, 100), 0)) {
+        TEST_info("OSSL_ECHSTORE_set1_key_and_readp_pem unexpected non-zero");
+        goto end;
+    }
+    /* this one fails 'cause priv has no real value, even if non NULL */
+    if (!TEST_int_eq(OSSL_ECHSTORE_set1_key_and_read_pem(es, priv, inout,
+                                                         OSSL_ECH_NO_RETRY),
+                     0)) {
+        TEST_info("OSSL_ECHSTORE_set1_key_and_readp_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_read_pem(NULL, inout, OSSL_ECH_NO_RETRY), 0)) {
+        TEST_info("OSSL_ECHSTORE_read_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_read_pem(es, NULL, OSSL_ECH_NO_RETRY), 0)) {
+        TEST_info("OSSL_ECHSTORE_read_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_read_pem(es, inout, 100), 0)) {
+        TEST_info("OSSL_ECHSTORE_read_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_num_keys(NULL, &count), 0)) {
+        TEST_info("OSSL_ECHSTORE_num_keys unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_num_keys(es, NULL), 0)) {
+        TEST_info("OSSL_ECHSTORE_num_keys unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_flush_keys(NULL, 0), 0)) {
+        TEST_info("OSSL_ECHSTORE_flush_keys unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_flush_keys(es, -1), 0)) {
+        TEST_info("OSSL_ECHSTORE_flush_keys unexpected non-zero");
+        goto end;
+    }
+    OSSL_ECH_INFO_free(NULL, 100);
```

Comment:
If  this call with (NULL,100à is a test, please describe the goal

Reply:
basic NULL check is all, added a comment